addons/blocklydevtools/userscript.js

```diff
@@ -0,0 +1,402 @@
+export default async function ({ addon, console }) {
+    await addon.tab.waitForElement("svg.blocklySvg>g.blocklyWorkspace");
+    if (!Blockly) { //If the Blockly global is not exposed, the script cannot do anything useful.
+        return "Failed to find Blockly instance!";
+    }
+    var workspace = Blockly.getMainWorkspace(); //Get the main Blockly workspace
```

Comment:
you should use `addon.tab.getWorkspace()` instead

Reply:
and probably also `const Blockly = await addon.tab.getBlockly()` instead of waiting for an element then trying to find window.Blockly

Reply:
Ok. When I wrote the addon I wrote it to work on any instance of blockly, Scratch or not. (It worked on Scratch, TurboWarp and Google's blockly demo with no modification). I just sort of did a copypaste for the addon. :)

I'll try to rewrite it a bit to work better with ScratchAddons. (Or, at least, look neater.)

Reply:
Done